### ISSUE TITLE
Dense Pixel Wise Labelling

### ISSUE THREAD

**Issue by ankurhanda** [BVLC/caffe#845]
Hi all, 

I have just modified caffe to run for dense pixel labelling for my semantic scene understanding task. I am testing that on a very simplistic network with only one convolution layer followed by ReLU and then softmax i.e.

data -> convolution ->ReLU -> softmax

The ground truth labelled image has 23 classes, therefore, I set the number of filters in convolution layer to only 23. The input is depth image of size 320x240 and the output of the network is 23 channel 320x240 images. The softmax loss layer then picks the appropriate label to compare that to the GT and computes the loss function which is summed over not just batch images but also on the number of pixels in the image. My training prototxt files looks like this: 

==================== TRAIN PROTOTXT ==============================
name: "VaFRIC_quick_train"
layers {
  name: "VaFRIC"
  type: DATA
  top: "data"
  top: "label"
  data_param {
    source: "/home/workspace/code/AnnotationGT/"
    batch_size: 100
  }
}
layers {
  name: "conv1"
  type: CONVOLUTION
  bottom: "data"
  top: "conv1"
  blobs_lr: 1
  blobs_lr: 2
  convolution_param {
    num_output: 23
    pad: 2
    kernel_size: 5
    stride: 1
    weight_filler {
      type: "gaussian"
      std: 0.0001
    }
    bias_filler {
      type: "constant"
    }
  }
}
layers {
  name: "relu1"
  type: RELU
  bottom: "conv1"
  top: "conv1"
}
layers {
  name: "loss"
  type: SOFTMAX_LOSS
  bottom: "conv1"
  bottom: "label"
}

However, I have been observing that the loss function only goes upwards and there is no decrease in the loss even when I change the learning rates and other variables that affect the SGDSolver.  I modified the cifar10_solver.prototxt file to use the these new train and test files I created: 
# 
## The training protocol buffer definition

train_net: "VaFRIC_quick_train.prototxt"
## The testing protocol buffer definition

test_net: "VaFRIC_quick_test.prototxt"
## covering the full 10,000 testing images.

test_iter: 1
## Carry out testing every 500 training iterations.

test_interval: 500
## The base learning rate, momentum and the weight decay of the network.

base_lr: 0.01
momentum: 0.9
weight_decay: 0.004
## The learning rate policy

lr_policy: "fixed"
gamma: 0.0001
power: 0.75
## Display every 100 iterations

display: 100
## The maximum number of iterations

max_iter: 4000
## snapshot intermediate results

snapshot: 4000
snapshot_prefix: "VaFRIC_quick"
## solver mode: CPU or GPU

solver_mode: CPU

I have been using ~1200 images all labelled for training but I cannot quite figure out why the loss function is always increasing. My main questions are 

(a) Is this too simple a network to converge? 
(b) Are there any other layers should I add to change the output?
(c) Do I need more data?
(d) How to verify that the changes I have made to obtain the dense output are indeed correct?

Being very new I'm unclear how to proceed from here. I'd be very grateful if I could get some reviews and feedback on this.  Many thanks for your time and patience.

Kind Regards,
Ankur.
